crates/boundless-market/src/contracts/mod.rs

```diff
@@ -1030,86 +905,78 @@ pub mod test_utils {
 
-    impl TestCtx {
-        async fn deploy_contracts(
-            anvil: &AnvilInstance,
-            set_builder_id: Digest,
-            assessor_guest_id: Digest,
-        ) -> Result<(Address, Address, Address, Address)> {
-            let deployer_signer: PrivateKeySigner = anvil.keys()[0].clone().into();
-            let deployer_provider = Arc::new(
-                ProviderBuilder::new()
-                    .with_recommended_fillers()
-                    .wallet(EthereumWallet::from(deployer_signer.clone()))
-                    .on_builtin(&anvil.endpoint())
-                    .await
-                    .unwrap(),
-            );
-
-            // Deploy contracts
-            let verifier = deploy_mock_verifier(Arc::clone(&deployer_provider)).await?;
-            let set_verifier =
-                deploy_set_verifier(Arc::clone(&deployer_provider), verifier, set_builder_id)
-                    .await?;
-            let hit_points =
-                deploy_hit_points(&deployer_signer, Arc::clone(&deployer_provider)).await?;
-            let boundless_market = deploy_boundless_market(
-                &deployer_signer,
-                Arc::clone(&deployer_provider),
-                set_verifier,
-                hit_points,
-                assessor_guest_id,
-                None,
-            )
+    async fn deploy_contracts(
+        anvil: &AnvilInstance,
+        set_builder_id: Digest,
+        assessor_guest_id: Digest,
+    ) -> Result<(Address, Address, Address, Address)> {
+        let deployer_signer: PrivateKeySigner = anvil.keys()[0].clone().into();
+        let deployer_provider = ProviderBuilder::new()
+            .wallet(EthereumWallet::from(deployer_signer.clone()))
+            .on_builtin(&anvil.endpoint())
             .await?;
 
-            // Mine forward some blocks using the provider
-            deployer_provider.anvil_mine(Some(U256::from(10)), Some(U256::from(2))).await.unwrap();
-            deployer_provider.anvil_set_interval_mining(2).await.unwrap();
+        // Deploy contracts
+        let verifier = deploy_mock_verifier(&deployer_provider).await?;
+        let set_verifier =
+            deploy_set_verifier(&deployer_provider, verifier, set_builder_id).await?;
+        let hit_points = deploy_hit_points(&deployer_signer, &deployer_provider).await?;
+        let boundless_market = deploy_boundless_market(
+            &deployer_signer,
+            &deployer_provider,
+            set_verifier,
+            hit_points,
+            assessor_guest_id,
+            None,
+        )
+        .await?;
 
-            Ok((verifier, set_verifier, hit_points, boundless_market))
-        }
+        // Mine forward some blocks using the provider
+        deployer_provider.anvil_mine(Some(10), Some(2)).await.unwrap();
+        deployer_provider.anvil_set_interval_mining(2).await.unwrap();
+
+        Ok((verifier, set_verifier, hit_points, boundless_market))
+    }
+
+    pub struct DefaultTestCtx;
```

Comment:
This struct appears to exist just to group together `DefaultTestCtx::create` and `DefaultTestCtx::create_with_rpc_url`, and IIUC these functions are not on TextCtx because of the way generics would force the type to be specified. I'd recommend that the these functions just be at the module level instead, so `create_test_ctx` and `create_text_ctx_with_rpc_url` instead. It's a bit awkward, but these isn't really an advantage to adding this struct here.